lib/llvm-backend/src/code.rs

```diff
@@ -1475,7 +1495,17 @@ impl<'ctx> FunctionCodeGenerator<CodegenError> for LLVMFunctionCodeGenerator<'ct
                         phi.add_incoming(&[(&value, current_block)]);
                     }
 
-                    builder.build_unconditional_branch(*frame.code_after());
+                    if current_block.get_first_instruction().is_none()
```

Comment:
Can you add a comment before this statement to give context?

```suggestion
                // We remove unneeded basic blocks, in case is empty
                if current_block.get_first_instruction().is_none()
```